pkg/manager/internal.go

```diff
@@ -265,8 +278,18 @@ func (cm *controllerManager) Start(stop <-chan struct{}) error {
 
 	go cm.startNonLeaderElectionRunnables()
 
-	if cm.resourceLock != nil {
-		err := cm.startLeaderElection()
+	if resourceLock := cm.resourceLocksMap[cm.leaderElectionID]; resourceLock != nil {
+		err := cm.startLeaderElection(cm.leaderElectionID, leaderelection.LeaderCallbacks{
+			OnStartedLeading: func(_ context.Context) {
+				cm.startLeaderElectionRunnables()
```

Comment:
I don't think we need this double-leader election here -- in fact, it kinda defeats the purpose of per-controller leader election.

Reply:
Really, you want to try to start all of the locks independently.